fulp_modules/features/species/beefmen/mob/beefman.dm

```diff
@@ -247,29 +265,50 @@
-	if(affecting || !istype(meat_slab, /obj/item/food/meat/slab))
+/datum/species/beefman/spec_attacked_by(var/obj/item/meat, mob/living/user, obj/item/bodypart/affecting, mob/living/carbon/human/beefboy)
+	if(affecting && !istype(meat, /obj/item/organ/tongue/beefman) && !istype(meat, /obj/item/food/meat/slab))
 		return ..()
 	var/target_zone = user.zone_selected
+	var/F = FALSE
 	if(!(target_zone in tearable_limbs))
 		return FALSE
-
-	user.visible_message(
-		span_notice("[user] begins mashing [meat_slab] into [beefboy]'s torso."),
-		span_notice("You begin mashing [meat_slab] into [beefboy]'s torso."))
-	// Leave Melee Chain (so deleting the meat doesn't throw an error) <--- aka, deleting the meat that called this very proc.
-	if(!do_mob(user, beefboy))
-		return FALSE
-	// Attach the part!
-	var/obj/item/bodypart/new_bodypart = beefboy.newBodyPart(target_zone, FALSE)
-	beefboy.visible_message(
-		span_notice("The meat sprouts digits and becomes [beefboy]'s new [new_bodypart.name]!"),
-		span_notice("The meat sprouts digits and becomes your new [new_bodypart.name]!"))
-	new_bodypart.attach_limb(beefboy)
-	new_bodypart.give_meat(beefboy, meat_slab)
-	qdel(meat_slab)
-	playsound(get_turf(beefboy), 'fulp_modules/features/species/sounds/beef_grab.ogg', 50, 1)
-	return TRUE
+	if(target_zone == BODY_ZONE_PRECISE_MOUTH && istype(meat, /obj/item/organ/tongue/beefman))
+		var/obj/item/organ/tongue/tongue = user.getorgan(/obj/item/organ/tongue)
+		if(tongue)
+			user.visible_message(
+				span_notice("You already have a tongue"))
+			return FALSE
+		user.visible_message(
+			span_notice("[user] begins mashing the Meaty tongue into [beefboy]'s mouth."),
+			span_notice("You begin mashing the Meaty tongue into [beefboy]'s mouth."))
+		if(!do_mob(user, beefboy))
+			return FALSE
+		user.visible_message(
+			span_notice("The meat sprouts and becomes [beefboy]'s new tongue!"),
+			span_notice("The tongue successfully fuses with your mouth!"))
+		var/obj/item/organ/tongue/beefman/new_tongue
+		new_tongue = new()
+		new_tongue.Insert(user, special = TRUE)
+		F = TRUE
+	if(target_zone !=BODY_ZONE_PRECISE_MOUTH && istype(meat, /obj/item/food/meat/slab))
+		user.visible_message(
+			span_notice("[user] begins mashing [meat] into [beefboy]'s torso."),
+			span_notice("You begin mashing [meat] into [beefboy]'s torso."))
+	    // Leave Melee Chain (so deleting the meat doesn't throw an error) <--- aka, deleting the meat that called this very proc.
+		if(!do_mob(user, beefboy))
+			return FALSE
+	    // Attach the part!
+		var/obj/item/bodypart/new_bodypart = beefboy.newBodyPart(target_zone, FALSE)
+		beefboy.visible_message(
+			span_notice("The meat sprouts digits and becomes [beefboy]'s new [new_bodypart.name]!"),
+			span_notice("The meat sprouts digits and becomes your new [new_bodypart.name]!"))
+		new_bodypart.attach_limb(beefboy)
+		new_bodypart.give_meat(beefboy, meat)
+		F = TRUE
+	if(F)
```

Comment:
I just want to make it clear, I hate this var.

Reply:
> I just want to make it clear, I hate this var.

its a MATHAfakin FLAG